opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -476,6 +475,58 @@ func resourceOpennebulaVirtualMachineUpdate(d *schema.ResourceData, meta interfa
 			if err != nil {
 				return fmt.Errorf("vm disk attach: %s", err)
 			}
+			d.SetPartial("disk")
+		}
+	}
+
+	if d.HasChange("nic") {
+
+		log.Printf("[INFO] Update NIC configuration")
+
+		old, new := d.GetChange("nic")
+		attachedNicsCfg := old.([]interface{})
+		newNicsCfg := new.([]interface{})
+
+		timeout := d.Get("timeout").(int)
+
+		// get the list of nics ID to detach
+		toDetach := diffIDsConfig(attachedNicsCfg, newNicsCfg, "network_id")
+
+		// Detach the nics
+		for _, nicIf := range toDetach {
+			nicConfig := nicIf.(map[string]interface{})
+
+			nicID := nicConfig["nic_id"].(int)
+
+			err := vmNICDetach(vmc, timeout, nicID)
+			if err != nil {
+				return fmt.Errorf("vm disk detach: %s", err)
+
+			}
+			d.SetPartial("nic")
+		}
+
+		// get the list of nics to attach
+		toAttach := diffIDsConfig(newNicsCfg, attachedNicsCfg, "network_id")
+
+		// Attach the nics
+		for _, nicIf := range toAttach {
+			nicConfig := nicIf.(map[string]interface{})
+
+			nicTpl := makeNICVector(map[string]interface{}{
```

Comment:
mac is still missing

Reply:
warning, all parameter excepted |network_id` are optional and must remains unchanged if there are not changed